ang/civicase/activity/card/directives/activity-card.directive.js

```diff
@@ -41,17 +41,29 @@
 
   module.controller('caseActivityCardController', caseActivityCardController);
 
-  function caseActivityCardController ($scope, dialogService, crmApi, crmBlocker, crmStatus, DateHelper, ts) {
+  /**
+   *
+   * @param {object} $scope scope object of the controller
+   * @param {object} dialogService service to open the dialog box
+   * @param {Function} crmApi service to interact with the civicrm api
+   * @param {object} crmBlocker crm blocker service
+   * @param {object} crmStatus crm status service
+   * @param {object} DateHelper date helper service
+   * @param {object} ts ts service
+   * @param {Function} viewInPopup factory to view an activity in a popup
+   */
+  function caseActivityCardController ($scope, dialogService, crmApi, crmBlocker, crmStatus, DateHelper, ts, viewInPopup) {
     $scope.ts = ts;
     $scope.formatDate = DateHelper.formatDate;
 
     /**
      * Mark an activity as complete
      *
-     * @param {object} activity
+     * @param {object} activity activity object
+     * @returns {Promise} api call promise
      */
     $scope.markCompleted = function (activity) {
-      return crmApi([['Activity', 'create', {id: activity.id, status_id: activity.is_completed ? 'Scheduled' : 'Completed'}]])
+      return crmApi([['Activity', 'create', { id: activity.id, status_id: activity.is_completed ? 'Scheduled' : 'Completed' }]])
```

Comment:
it would be good to define the API call outside of this line to avoid having a long line with logic that is difficult to follow.

Reply:
I will do this next time :|